inputremapper/injection/injector.py

```diff
@@ -348,9 +348,6 @@ def run(self) -> None:
 
         try:
             loop.run_until_complete(asyncio.gather(*coroutines))
-        except RuntimeError:
-            # stopped event loop most likely
-            pass
```

Comment:
Was there a good reason to have this?

Reply:
might be `quick_cleanup`.

Also, there happens to be another `except RuntimeError` in `quick_cleanup`, with the explanation `# happens when the event loop disappears for magical reasons`. So I probably never solved the mystery why this happens, and I think it only happens in tests.

Why are you removing it?

Reply:
Nevermind, I know now why Unfortunately it catches to many errors. we should probably at least print something to the debug log

Reply:
I think it might be important to catch exceptions here in order to ungrab the devices properly before finishing with `run`.

Consider something like

```py
        except RuntimeError as error:
            # stopped event loop most likely
            logger.error(error)
            pass
```

Reply:
stopping the event loop from the _msg_listener() causes the exception.
maybe there is a way to cancel all futures when the _msg_listener() receives CLOSE, then it should stop automatically.
for now I do it like this:
```
 try:
    loop.run_until_complete(asyncio.gather(*coroutines))
except RuntimeError as error:
    if error.__str__() != "Event loop stopped before Future completed.":
        raise error
except OSError as error:
    logger.error("Failed to run injector coroutines: %s", str(error))
```

Reply:
```py
 try:
    loop.run_until_complete(asyncio.gather(*coroutines))
except RuntimeError as error:
    # the loop might have been stopped via a `CLOSE` message,
    # which causes the error message below. This is expected behavior
    if str(error) != "Event loop stopped before Future completed.":
        raise error
except OSError as error:
    logger.error("Failed to run injector coroutines: %s", str(error))
```